Tests/CDBElevationTest.cpp

```diff
@@ -619,3 +643,108 @@ TEST_CASE("Test that elevation conversion uses uniform grid mesh instead of simp
+                // Check all enums are written.
+                auto enumValues = gltf.extensions["EXT_feature_metadata"].Get("schema").Get("enums").Get("CDBBaseMaterial").Get("values");
+                REQUIRE(enumValues.ArrayLen() == 131);
+            }
+        }
+    }
+
+    // remove the test output
+    std::filesystem::remove_all(output);
+}
+
+TEST_CASE("Test writing of external schema for RMDescriptor", "[3D Tiles Next]")
+{
+    std::filesystem::path input = dataPath / "ElevationWithRMTextureRMDescriptor";
+    std::filesystem::path output = dataPath / "ElevationWithRMTextureRMDescriptorOutput";
+    std::filesystem::path elevationOutputDir = output / "Tiles" / "N12" / "E044" / "Elevation" / "1_1";
+    std::filesystem::path textureOutputDir = elevationOutputDir / "Textures";
+
+    Converter converter(input, output);
+    converter.setUse3dTilesNext(true);
+    converter.setExternalSchema(true);
+    converter.convert();
+
+    // Check feature ID texture generation
+    checkAllConvertedRMTexture(input / "Tiles" / "N12" / "E044" / "005_RMTexture", textureOutputDir, 4);
+
+    // Check feature tables creation in glTFs.
+    tinygltf::TinyGLTF gltfIO;
+    tinygltf::Model gltf;
+    std::string error, warning;
+    std::filesystem::path elevationPath = input / "Tiles" / "N12" / "E044" / "001_Elevation";
+    for (std::filesystem::directory_entry levelDir : std::filesystem::directory_iterator(elevationPath)) {
+        for (std::filesystem::directory_entry UREFDir : std::filesystem::directory_iterator(levelDir)) {
+            for (std::filesystem::directory_entry tilePath : std::filesystem::directory_iterator(UREFDir)) {
+                auto tileFilename = tilePath.path();
+                if (tileFilename.extension() != ".tif") {
+                    continue;
+                }
+
+                std::filesystem::path glbPath = elevationOutputDir / (tileFilename.stem().string() + ".glb");
+                REQUIRE(std::filesystem::exists(glbPath));
+
+                gltfIO.LoadBinaryFromFile(&gltf, &error, &warning, glbPath, 0);
+
+                REQUIRE(
+                    std::find(gltf.extensionsUsed.begin(), gltf.extensionsUsed.end(), "EXT_feature_metadata")
+                    != gltf.extensionsUsed.end());
+                REQUIRE(gltf.extensions.find("EXT_feature_metadata") != gltf.extensions.end());
+                
+                // Check all enums are written.
+                REQUIRE(gltf.extensions["EXT_feature_metadata"].Has("schemaUri"));
+            }
+        }
+    }
+
+    // Check if schema JSON is written.
+    std::filesystem::path materialsSchemaPath = output / "materials.json";
+    REQUIRE(std::filesystem::exists(materialsSchemaPath));
+
+    // Check if all base materials are written.
+    std::ifstream schemaStream(materialsSchemaPath);
+    nlohmann::json materialsSchema = nlohmann::json::parse(schemaStream);
+    REQUIRE(materialsSchema["enums"]["CDBBaseMaterial"]["values"].size() == 131);
```

Comment:
Can this 131 be defined outside of both test cases, and then both refer to the one variable instead of having a wild 131?